ios/Classes/PspdfkitPlugin.m

```diff
@@ -41,23 +46,130 @@ - (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
+# pragma mark - Private methods
+
+- (PSPDFDocument *)document:(NSString *)path {
     NSURL *url;
     
-    if ([string hasPrefix:@"/"]) {
-        url = [NSURL fileURLWithPath:string];
+    if ([path hasPrefix:@"/"]) {
+        url = [NSURL fileURLWithPath:path];
     } else {
-        url = [NSBundle.mainBundle URLForResource:string withExtension:nil];
+        url = [NSBundle.mainBundle URLForResource:path withExtension:nil];
     }
-
-    NSString *fileExtension = url.pathExtension.lowercaseString;
-    BOOL isImageFile = [fileExtension isEqualToString:@"png"] || [fileExtension isEqualToString:@"jpeg"] || [fileExtension isEqualToString:@"jpg"];
-    if (isImageFile) {
+    
+    if ([self isImageDocument:path]) {
         return [[PSPDFImageDocument alloc] initWithImageURL:url];
     } else {
         return [[PSPDFDocument alloc] initWithURL:url];
     }
 }
 
+- (PSPDFConfiguration *)configuration:(NSDictionary *)dictionary isImageDocument:(BOOL)isImageDocument {
+    PSPDFConfiguration *configuration;
+    if (isImageDocument) {
+        configuration = PSPDFConfiguration.imageConfiguration;
+    } else {
+        configuration = PSPDFConfiguration.defaultConfiguration;
+    }
+    
+    if ((id)dictionary == NSNull.null || !dictionary || dictionary.count == 0) {
+        return configuration;
+    }
+    
+    return [configuration configurationUpdatedWithBuilder:^(PSPDFConfigurationBuilder * _Nonnull builder) {
+        if (dictionary[@"pageScrollDirection"]) {
+            builder.scrollDirection = [dictionary[@"pageScrollDirection"] isEqualToString:@"pageScrollDirectionHorizontal"] ? PSPDFScrollDirectionHorizontal : PSPDFScrollDirectionVertical;
+        }
+        if (dictionary[@"pageScrollContinuous"]) {
+            builder.pageTransition = dictionary[@"pageScrollContinuous"] ? PSPDFPageTransitionScrollContinuous : PSPDFPageTransitionScrollPerSpread;
+        }
+        if (dictionary[@"userInterfaceViewMode"]) {
+            builder.userInterfaceViewMode = [self userInterfaceViewMode:dictionary[@"userInterfaceViewMode"]];
+        }
+        if (dictionary[@"inlineSearch"]) {
+            builder.searchMode = dictionary[@"inlineSearch"] ? PSPDFSearchModeInline : PSPDFSearchModeModal;
+        }
+        if (dictionary[@"showThumbnailBar"]) {
+            builder.thumbnailBarMode = [self thumbnailBarMode:dictionary[@"showThumbnailBar"]];
+        }
+        if (dictionary[@"showPageLabels"]) {
+            builder.pageLabelEnabled = [dictionary[@"showPageLabels"] boolValue];
+        }
+        if (![dictionary[@"enableAnnotationEditing"] boolValue]) {
+            builder.editableAnnotationTypes = nil;
+        }
+        if (dictionary[@"enableTextSelection"]) {
+            builder.textSelectionEnabled = [dictionary[@"enableTextSelection"] boolValue];
+        }
+    }];
+}
+
+# pragma mark - Helpers
+
+- (PSPDFUserInterfaceViewMode)userInterfaceViewMode:(NSString *)stringValue {
```

Comment:
I think it would be nicer if we would provide the whole dictionary as the parameter, and the method handles extracting it, so the caller doesn't need to know the actual key for user interface view mode in the dictionary.
Also it would fit the already existing methods, that all take a dictionary.

Reply:
I like this. However, I would still do the check for the key in `-configuration:isImageDocument:` like this:

```objc
...
if (dictionary[@"showThumbnailBar"]) {
    builder.thumbnailBarMode = [self thumbnailBarMode:dictionary];
}
...
```

I'd do this to avoid setting the default value when not explicitly specified.

Reply:
Do you feel like there is any harm in setting the default value when not specified? I think it should be fine to just return and set the default value.
We could use `PSPDFConfiguration.defaultConfiguration.userInterfaceViewMode` to automatically get the default without having to hardcode it here though, to automatically get the SDK default.

Reply:
> Do you feel like there is any harm in setting the default value when not specified?

I was thinking of the optimization of not setting it again to the value, but I agree that we should be fine to just return and set the default value.

> We could use PSPDFConfiguration.defaultConfiguration.userInterfaceViewMode to automatically get the default without having to hardcode it here though, to automatically get the SDK default.

Makes total sense as the SDK's default configuration values may change and we shouldn't hardcode it.

Addressed in https://github.com/PSPDFKit/pspdfkit-flutter/pull/10/commits/cda75e42334632987516447aec51c87f1e5d75f0